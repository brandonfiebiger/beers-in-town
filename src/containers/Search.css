.location-inputs {
  height: 2.56rem;
  width: 10rem;
  margin-right: 1.2rem;
  border-radius: 10px;
  background: rgba(25, 25, 25, .5);
  color: #f9f9f9;
  font-size: 1.2rem;
}

.location-inputs::placeholder {
  color: #f9f9f9;
}

.search-button {
  height: 3rem;
  width: 10rem;
  border-radius: 10px;
  font-size: 1.2rem;
  background: rgba(25, 25, 25, .9);
  cursor: pointer;
  color: #f9f9f9;
}

.search-button:hover {
  background: rgba(25, 25, 25, 1);

}


@media screen and (max-width: 560px) {

  form {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .location-inputs {
    margin-right: 0;
    margin-bottom: 0.5rem;
  }
}