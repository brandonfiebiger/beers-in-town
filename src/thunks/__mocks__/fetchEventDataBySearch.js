export const fetchEventDataBySearch = jest.fn()