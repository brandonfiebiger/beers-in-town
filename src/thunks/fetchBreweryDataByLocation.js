import { apikey } from '../utils/variables'
import { populateBreweriesFromLocation } from '../actions/';

export const fetchBreweryDataByLocation = (latitude, longitude) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://api.meetup.com/find/venues?key=${apikey}&sign=true&photo-host=secure&text=brewery&lon=${longitude}&radius=100&lat=${latitude}&page=200`);
      //add condition to check that response is ok check thunk lesson
      const breweries = await response.json();
      dispatch(populateBreweriesFromLocation(breweries))
    } catch (error) {
       alert('could not get your location')
    }
  }
}
