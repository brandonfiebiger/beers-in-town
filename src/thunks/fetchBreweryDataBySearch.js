import { apikey } from '../utils/variables'
import { populateBreweriesFromSearch } from '../actions/';

export const fetchBreweryDataBySearch = (city, state) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://api.meetup.com/2/open_venues?key=${apikey}&sign=true&photo-host=public&country=us&city=${city}&state=${state}&text=brewery+craft+beer&page=100`);
      const breweries = await response.json();
      dispatch(populateBreweriesFromSearch(breweries.results))
    } catch (error) {
       throw new Error(error.message)
    }
  }
}
