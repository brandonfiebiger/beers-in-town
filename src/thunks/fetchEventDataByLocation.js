import { populateEventsFromLocation } from '../actions/';
import { apikey } from '../utils/variables';
import { eventCleaner } from '../utils/helper';

export const fetchEventDataByLocation = (latitude, longitude) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://api.meetup.com/find/upcoming_events?key=${apikey}&sign=true&photo-host=public&lon=${longitude}&page=2000&text=craft beer, brewery&radius=100&lat=${latitude}`);
      const events = await response.json();
      const cleanEvents = await eventCleaner(events.events);
      dispatch(populateEventsFromLocation(cleanEvents))
    } catch (error) {
       alert('Could not get your location')
    }
  }
}


