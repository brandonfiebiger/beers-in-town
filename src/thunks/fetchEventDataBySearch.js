import { populateEventsFromSearch } from '../actions/';
import { apikey } from '../utils/variables';
import { eventCleaner } from '../utils/helper';

export const fetchEventDataBySearch = (city, state) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://api.meetup.com/2/open_events?key=766f10297e73965263a3b52f693770&sign=true&photo-host=public&country=us&city=${city}&state=${state}&text=brewery&page=100`);
      const events = await response.json();
      const cleanEvents = await eventCleaner(events.results);
      console.log(cleanEvents)
      dispatch(populateEventsFromSearch(cleanEvents))
    } catch (error) {
       throw new Error(error.message)
    }
  }
}


