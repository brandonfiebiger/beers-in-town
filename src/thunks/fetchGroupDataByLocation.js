import { apikey } from '../utils/variables';
import  { populateGroupsFromLocation } from '../actions/';
import { cleanGroupData } from '../utils/helper';


export const fetchGroupDataByLocation = (latitude, longitude) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://api.meetup.com/find/groups?key=${apikey}&sign=true&photo-host=public&lon=${longitude}&text=craft beer&radius=100&lat=${latitude}&page=100`);
      const groups = await response.json();
      const cleanedGroups = await cleanGroupData(groups);
      dispatch(populateGroupsFromLocation(cleanedGroups))
    } catch (error) {
      throw new Error(error.message)
    }
  }
}

