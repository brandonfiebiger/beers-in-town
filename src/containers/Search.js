import React, { Component } from 'react';
import { fetchEventDataBySearch } from '../thunks/fetchEventDataBySearch';
import { fetchGroupDataBySearch } from '../thunks/fetchGroupDataBySearch';
import { fetchBreweryDataBySearch } from '../thunks/fetchBreweryDataBySearch';
import { Route, withRouter, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';


export class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      city: '',
      state: ''
    }
  }

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({
      [ name ]: value
    })
  }

  handleSubmit = event => {
    event.preventDefault()
    const { city, state } = this.state
    this.props.getEvents(this.state.city, this.state.state)
    this.props.getGroups(this.state.city, this.state.state)
    this.props.getBreweries(this.state.city, this.state.state)
    this.props.history.push('/events')
  }


  render() {
    return(
      <form onSubmit={ this.handleSubmit } >
        <input name='city' value={this.state.city} onChange={this.handleChange}/>
        <input name='state' value={this.state.state} onChange={this.handleChange}/>
        <button>Search</button>
      </form>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  getEvents: (city, state) => dispatch(fetchEventDataBySearch(city, state)),
  getGroups: (city, state) => dispatch(fetchGroupDataBySearch(city, state)),
  getBreweries: (city, state) => dispatch(fetchBreweryDataBySearch(city,state))
})

export default withRouter(connect(null, mapDispatchToProps)(Search));

Search.propTypes = {
  getEvents: PropTypes.func,
  getGroups: PropTypes.func,
  getBreweries: PropTypes.func
};