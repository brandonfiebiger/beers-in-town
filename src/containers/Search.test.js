import React from 'react';
import ReactDOM from 'react-dom';
import { Search, mapDispatchToProps } from './Search';
import { shallow, mount } from 'enzyme';
import { fetchEventDataBySearch } from '../thunks/fetchEventDataBySearch';


describe('Search', () => {
  let mockHistory;
  let wrapper;
  let event
  beforeEach(() => {
    event = {
      preventDefault: jest.fn(),
      target: {
        value: 'hello',
        name: 'name',
      },
    };
    mockHistory = { push: jest.fn() };
    wrapper = shallow(<Search getEvents={ jest.fn() } getGroups={ jest.fn() } getBreweries={ jest.fn() } history={ mockHistory }/>)
  })


  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  // it('should call getEvents, getBreweries and getGroups when handleSubmit is invoked', () => {
  //   wrapper = shallow(<Search getEvents={ jest.fn() } getGroups={ jest.fn() } getBreweries={ jest.fn() } history={ mockHistory }/>)

  //   wrapper.state().city = 'Arlen';
  //   wrapper.state().state = 'Texas';
  //   wrapper.instance().handleSubmit(event)
  //   console.log(wrapper.instance().getBreweries)
  //   expect().toHaveBeenCalledWith()
  // })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a fetchEventDataBySearch action', () => {
      const mockDispatch = jest.fn();

      const mockLocation = { city: 'Arlen', state: 'Texas' };
      const actionToDispatch = fetchEventDataBySearch('Arlen', 'Texas')
      
      const mappedProps = mapDispatchToProps(mockDispatch);
      console.log(mappedProps)

      mappedProps.getEvents('Arlen', 'Texas');

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)

    })
  })
})