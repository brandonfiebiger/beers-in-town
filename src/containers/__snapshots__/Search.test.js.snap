// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    getBreweries={[Function]}
    getEvents={[Function]}
    getGroups={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="city"
          onChange={[Function]}
          placeholder="city Ex. Madison"
          value=""
/>,
        <input
          name="state"
          onChange={[Function]}
          placeholder="state Ex. WI"
          value=""
/>,
        <button>
          Search
</button>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "city",
          "onChange": [Function],
          "placeholder": "city Ex. Madison",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "state",
          "onChange": [Function],
          "placeholder": "state Ex. WI",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search",
        },
        "ref": null,
        "rendered": "Search",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="city"
            onChange={[Function]}
            placeholder="city Ex. Madison"
            value=""
/>,
          <input
            name="state"
            onChange={[Function]}
            placeholder="state Ex. WI"
            value=""
/>,
          <button>
            Search
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "city",
            "onChange": [Function],
            "placeholder": "city Ex. Madison",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "state",
            "onChange": [Function],
            "placeholder": "state Ex. WI",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
